src/nvim/message.c

```diff
@@ -2478,8 +2471,10 @@ static void redir_write(char_u *str, int maxlen)
       }
     }
 
-    if (redir_reg) {
-      size_t len = maxlen == -1 ? STRLEN(s) : (size_t)maxlen;
+    size_t len = maxlen == -1 ? STRLEN(s) : (size_t)maxlen;
+    if (capture_ga) {
+      ga_concat_len(capture_ga, (const char *)str, len);
+    } else if (redir_reg) {
```

Comment:
I would think this should be just "if" instead of "else if", to support nesting, no?